### PR TITLE
[WIP] Collections

### PR DESCRIPTION
**Pull request by rougier**
~~This needs PR #765 to be merge before going further.~~

### CODE REVIEW

File: vispy/visuals/collections/base_collection.py [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+            for program in self._programs:
+                program.bind(self._vertices_buffer)
+                if self._uniforms_list is not None:
+                    program["uniforms"] = self._uniforms_texture
+                    program["uniforms_shape"] = self._ushape
+
+
+
+# -----------------------------------------------------------------------------
+if __name__ == '__main__':
```
Comment (campagnola) [inline vispy/visuals/collections/base_collection.py]: Turn this into a unit test?

File: vispy/visuals/collections/raw_point_collection.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        Initialize the collection.
+
+        Parameters
+        ----------
+
+        user_dtype: list
+            The base dtype can be completed (appended) by the used_dtype. It
+            only make sense if user also provide vertex and/or fragment shaders
+
+        viewport: glumpy.Transforms
```
Comment (campagnola) [inline vispy/visuals/collections/raw_point_collection.py]: Need to do something with this..   maybe make it into a visual?

File: vispy/visuals/collections/util.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            count -= i
+            shift += i
+            store -= i
+
+    body += """}\n\n"""
+    return header + body
+
+
+# -----------------------------------------------------------------------------
+if __name__ == '__main__':
```
Comment (campagnola) [inline vispy/visuals/collections/util.py]: unit test here as well.

**Comment by rossant** [comment]
TODO
* Transforms & viewport
* Python 3 fixes
* flakify

**Comment by rougier** [comment]
This PR is totally messed up... We'll use Cyril's PR

**Pull request closed without merging**